### PR TITLE
AKU-974: Visibility Config

### PR DESCRIPTION
**Pull request by draperd**
This PR addresses https://issues.alfresco.com/jira/browse/AKU-974 / #1047 to support state based publication evaluation for non-form controls. 

The new changes mean that it is possible to use the form style visibility configuration that persists the state changes of the various topics. It is also possible to mix and match topics and target ids as well as configure the attribute used for the value. Unit tests have been added to try and cover all the various use cases.

### CODE REVIEW

File: aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js [hunk 1]
```diff
@@ -308,6 +309,25 @@ define(["dojo/_base/declare",
...
+      _visibilitySetting: function alfresco_core_CoreWidgetProcessing___visibilitySetting(status, widget) {
+         if (status)
+         {
+            domStyle.set(widget.domNode, "display", "");
+         }
+         else
+         {
+            domStyle.set(widget.domNode, "display", "none");
+         }
+      },
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js]: This'd be a lot shorter with a ternary expression...

```javascript
domStyle.set(widget.domNode, "display", status ? "" : "none");
```

File: aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js [hunk 2]
```diff
@@ -328,38 +348,47 @@ define(["dojo/_base/declare",
          // as requested and then set up the subcriptions...
          if (widget[configAttribute])
          {
-            var initialValue = lang.getObject(configAttribute + ".initialValue", false, widget);
-            initialValue = negate ? !initialValue : initialValue;
-            if (initialValue != null && initialValue === false)
+            // See AKU-974 - support for "form" style visibility config...
+            var useState = lang.getObject(configAttribute + ".useState", false, widget);
+            if (useState)
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js]: I presume falsy values here won't cause any issues?

File: aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js [hunk 3]
```diff
@@ -328,38 +348,47 @@ define(["dojo/_base/declare",
...
-                           strict: strict
-                        },
-                        successCallback = lang.hitch(this, this.onVisibilityProcessedSuccess, widget),
-                        failureCallback = lang.hitch(this, this.onVisibilityProcessedFailure, widget);
-                     widget.alfConditionalSubscribe(topic, rulesObj, successCallback, failureCallback);
-                  }
-               }, this);
+               var initialValue = lang.getObject(configAttribute + ".initialValue", false, widget);
+               initialValue = negate ? !initialValue : initialValue;
+               if (initialValue != null && initialValue === false)
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js]: Don't need the first condition with a strict equals as the second.

File: aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js (hunk 2 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js]: No - plenty of test cases proving that.

**Pull request merged**